### DISCUSSION TITLE
How can i do file logging in bitPlatform ?

### DISCUSSION THREAD

**Discussion by HarminderArtesianIo** [bitfoundation/bitplatform#8810] [Q&A]
I have a webapi that needs file logging, _please share working example code_.

**Comment by HarminderArtesianIo** [comment]
Note: Currently my api backend application is writing logs on console only.

Also share command/instruction to write my own logs in controllers/other classes.

**Comment by yasmoradi** [comment]
You can install any Microsoft extension logging based log provider, what's the issue?

**Reply by HarminderArtesianIo** [reply]
**Scenario:**
I am writing logs using Serilog to both Console and Files and it is working.
There are logs, that are appearing in the console , not written by me. BitPlatform app is writting them. 
**My Requirement is:**
a. I want to write all the logs in the file, only serilogs are going to file.
b. and find from where non-serilog are configured in the app? I am using the bitPlatform template, that has logging already working.

<img width="524" alt="image" src="https://github.com/user-attachments/assets/25c8b69d-0b5c-4840-82b8-5fc56186f771">
<img width="567" alt="image" src="https://github.com/user-attachments/assets/049f660f-e8c9-4ddc-b8b9-f4682d28b3e8">

**Reply by yasmoradi** [reply]
1- Remove Serilog
2- Install and configure Karambolo.Extensions.Logging.File
3- Use ILogger to write in all registered Microsoft.Extensions.Logging based log providers including Karambolo.Extensions.Logging.File

**Reply by HarminderArtesianIo** [reply]
please share working example code.

**Reply by yasmoradi** [reply]
I'm afraid, but that's not something we do offer for non-paid users.
You can receive lots of benefits including code-review, pair sessions, faster replies, custom made solution etc by [purchasing our support package](https://bitplatform.dev/pricing)
